enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,244 @@
+  // be block
+  // + optional
+  VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+  // FSType type to create when volumeMode is Filesystem
+  // +optional
+  FSType string `json:"fsType,omitempty"`
+  // If specified, a list of tolerations to pass to the discovery daemons.
+  // +optional
+  Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+  // DeviceInclusionSpec is the filtration rule for including a device in the device discovery
+  // +optional
+  DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+  
+  // PartitioningSpec is the partitioning policy according to which the selected devices will be partitioned.
+  PartitioningSpec PartitoiningSpec `json:"partitioningSpec"`
+}
+
+type PartitoiningSpec struct {
+  // PartSize determines the exact size of each partition.
+  PartSize *resource.Quantity `json:"partSize"`
+}
+
+type LocalVolumeSetStatus struct {
+  // Conditions is a list of conditions and their status.
+  Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+  
+  // TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+  TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+  
+  // observedGeneration is the last generation change the operator has dealt with
+  // +optional
+  ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+  
+  // TotalProvisionedPartitionCount is the count of the total partitions over which the PVs has been provisioned
+  // This will be diffrent for TotalProvisionedDeviceCount as TotalProvisionedDeviceCount is
+  //the count of the actual devices
+  TotalProvisionedPartitionCount *int32 `json:"totalProvisionedPartitionCount,omitempty"`
+  
+  // PartitioningStatus will track the status of partitioning
+  // Valid values are
+  // nil : if there is no partitioningSpec
+  // partitoining : if the partitioning is in progress
+  // completed: when the partitioning of devices is completed
+  PartitioningStatus string `json:"partitioningStatus"`
+}
+```
+
+### Workflow of `LocalPV` creation via `LocalVolumeSet CR`
+- Once the `LocalVolumeSet` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemonset to the selected nodes.
+  - there will be one daemonset for each `LocalVolumeSet` CR.
+- The diskmaker daemon will find devices that match the disovery policy and check if partitoningSpec is present
+  - if partitonSpec is present it will set PartitioningStatus to partitioning and will partition the devices.
+  - as soon as a device is partitoned
+    - the diskmaker will symlink the child partitions of the device into the directory that the local-static-provisioner is watching
+    - will set partitioningStatus to Completed once all the devices are partitioned
```

Comment:
Only the operator should update the CR status, otherwise each of the nodes will conflict when they update the status.